src/main/scala/com/databricks/spark/sql/perf/query.scala

```diff
@@ -36,7 +36,7 @@ case class QueryForTest(
     (endTime - startTime).toDouble / 1000000
   }
 
-  def benchmark(description: String = "") = {
+  def benchmark(description: String = "", queryOutputLocation: Option[String]) = {
```

Comment:
Could we instead do something like the following?

```scala
abstract class ExecutionMode
case object CollectResults extends ExecutionMode
case object CountResults extends ExecutionMode
case object WriteParquet(location: String) extends ExecutionMode
```

This can replace the `collectResults` boolean and the query itself can tell the harness how it should be run.